src/main/java/org/biouno/unochoice/util/Utils.java

```diff
@@ -227,18 +234,91 @@ private static boolean isParameterDefintionOf(@Nonnull String parameterUUID, @No
+            for (NodeProperty<?> nodeProperty : globalNodeProperties) {
+                if (nodeProperty instanceof EnvironmentVariablesNodeProperty) {
+                    EnvironmentVariablesNodeProperty envNodeProperty = (EnvironmentVariablesNodeProperty) nodeProperty;
+                    map.putAll(envNodeProperty.getEnvVars());
                 }
             }
         }
         return map;
     }
+
+    /**
+     * Get parameter definitions associated with {@link BuildWrapper}s of the given {@link Project}.
+     * @param project the project for which the parameter definitions should be found
+     * @return Map
+     */
+    public static @Nonnull Map<BuildWrapper, List<ParameterDefinition>> getBuildWrapperParameterDefinitions(@Nonnull Project<?, ?> project) {
+        final List<BuildWrapper> buildWrappersList = project.getBuildWrappersList();
+
+        final Map<BuildWrapper, List<ParameterDefinition>> result = new LinkedHashMap<BuildWrapper, List<ParameterDefinition>>();
+
+        List<ParameterDefinition> value = new ArrayList<ParameterDefinition>();
+
+        for (BuildWrapper buildWrapper : buildWrappersList) {
+            final PropertyDescriptor[] propertyDescriptors;
+            try {
+                propertyDescriptors = Introspector.getBeanInfo(buildWrapper.getClass()).getPropertyDescriptors();
+            } catch (IntrospectionException e) {
+                continue;
+            }
+            for (PropertyDescriptor propertyDescriptor : propertyDescriptors) {
+                addParameterDefinitionsTo(value, buildWrapper, propertyDescriptor);
+            }
+            if (!value.isEmpty()) {
+                result.put(buildWrapper, value);
+                value = new ArrayList<ParameterDefinition>();
+            }
+        }
+        return result.isEmpty() ? Collections.<BuildWrapper, List<ParameterDefinition>> emptyMap() : result;
+    }
+
+    private static void addParameterDefinitionsTo(List<ParameterDefinition> target, Object bean, PropertyDescriptor pd) {
+        if (ParameterDefinition.class.isAssignableFrom(pd.getPropertyType())) {
+            final ParameterDefinition param = read(bean, pd);
+            if (param != null) {
+                target.add(param);
+            }
+            return;
+        }
+        Iterable<?> iterable = null;
+
+        if (Iterable.class.isAssignableFrom(pd.getPropertyType())) {
+            iterable = read(bean, pd);
+        } else if (Object[].class.isAssignableFrom(pd.getPropertyType())) {
+            final Object[] array = read(bean, pd);
+            if (array != null)
+                iterable = Arrays.asList(array);
+        }
+        if (iterable == null)
+            return;
+
+        for (Object o : iterable) {
+            if (ParameterDefinition.class.isInstance(o)) {
+                target.add((ParameterDefinition) o);
+            }
+        }
+    }
+
+    private static <T> T read(Object bean, PropertyDescriptor pd) {
```

Comment:
The only thing that I am not entirely sure in the PR is the name of this method. But I can't think of any better name... maybe `readParameterDefinitions`, or `getParameterDefinitions`, or `findParameterDefinitions` ?